libs/execution_context/ebpf_program.c

```diff
@@ -1084,30 +1084,57 @@ ebpf_program_detach_link(_Inout_ ebpf_program_t* program, _Inout_ ebpf_link_t* l
 ebpf_result_t
 ebpf_program_get_info(
     _In_ const ebpf_program_t* program,
-    _Out_writes_to_(*info_size, *info_size) uint8_t* buffer,
+    _In_reads_(*info_size) const uint8_t* input_buffer,
+    _Out_writes_to_(*info_size, *info_size) uint8_t* output_buffer,
     _Inout_ uint16_t* info_size)
 {
     EBPF_LOG_ENTRY();
-    struct bpf_prog_info* info = (struct bpf_prog_info*)buffer;
-    if (*info_size < sizeof(*info)) {
+    const struct bpf_prog_info* input_info = (const struct bpf_prog_info*)input_buffer;
+    struct bpf_prog_info* output_info = (struct bpf_prog_info*)output_buffer;
+    if (*info_size < sizeof(*output_info)) {
         EBPF_RETURN_RESULT(EBPF_INSUFFICIENT_BUFFER);
     }
 
-    info->id = program->object.id;
+    ebpf_result_t result = EBPF_SUCCESS;
+    if ((input_info->map_ids != 0) && (input_info->nr_map_ids > 0) && (program->count_of_maps > 0)) {
```

Comment:
maybe you could check that `nr_map_ids == count_of_maps` if `map_ids != NULL`?

Reply:
(Updated my comment above)
Line 1111 returns error if no more space left. I am simply asking to make that check prior to entering the loop.